packages/apps/graph/src/services/confirmation-depth-service.ts

```diff
@@ -0,0 +1,48 @@
+import { prismaClient } from '@db/prisma-client';
+
+export async function getLatestBlockHeights(chainIds?: string[]) {
+  return await prismaClient.block.groupBy({
+    by: ['chainId'],
+    _max: {
+      height: true,
+    },
+    where: {
+      chainId: {
+        in: (chainIds as string[]).map((id) => parseInt(id)),
+      },
+    },
+  });
+}
+
+export async function getConditionForMinimumDepth(
+  minimumDepth: number,
+  chainIds?: string[],
+): Promise<
+  {
+    chainId: bigint;
+    height: {
+      lte: number;
+    };
+  }[]
+> {
+  const latestBlocks = await prismaClient.block.groupBy({
+    by: ['chainId'],
+    _max: {
+      height: true,
+    },
+    where: {
+      chainId: {
+        in: (chainIds as string[]).map((id) => parseInt(id)),
+      },
+    },
```

Comment:
Only do this `where` if chainIds are passed.

```suggestion
    ...(chainIds?.length && { where: {
      chainId: {
        in: chainIds.map((id) => parseInt(id)),
      },
    }}),
```

Reply:
Good catch